File: Python/leetcode/p1.py
# https://leetcode.com/problems/two-sum/

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        looking_for = {}
        for i, num in enumerate(nums):
            if num in looking_for:
                return looking_for[num], i
            looking_for[target - num] = i


def test(*args):
    print(Solution().twoSum(*args))


test([2, 7, 11, 15], 9)

File: Rust/leetcode/p1.rs
// https://leetcode.com/problems/two-sum/

// impl Solution {
pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    use std::collections::HashMap;
    let mut looking_for: HashMap<i32, i32> = HashMap::new();

    for (i, num) in nums.iter().enumerate() {
        match looking_for.get(num) {
            None => looking_for.insert(target - num, i as i32),
            Some(&index) => return vec![index, i as i32 as i32],
        };
    }
    vec![]
}
// }
